### PR TITLE
Create script to precompute and persist HOC signup aggregates.

### PR DESCRIPTION
**Pull request by ashercodeorg**
This script is being created both to be run in an adhoc fashion for prior years (via the `ARGV[0]` variable) and in a daily (?) fashion for the current year (via the `DCDO.get('hoc_year', ...)`).

A later PR will convert our event pages (e.g., [hourofcode.com/events/2014](hourofcode.com/events/2014)) to use this persisted value rather than querying SOLR. This is an improvement, both dramatically reducing the pageload time and removing a SOLR dependency.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Note that this script does not currently dedupe events by email address. Should it? Do the existing SOLR numbers do so?

**Comment by ashercodeorg** [comment]
Also note that this script does not work for the splat pages, e.g., [https://hourofcode.com/us/events/2014/ca](https://hourofcode.com/us/events/2014/ca). Is there a product need for us to maintain them? If so, a DB-based solution will require my adding additional indexes to the `form_geos` table.

**Comment by tanyaparker** [comment]
I don't know if SOLR reports de-duped data... but it is my belief that any externally reported data should be de-duped, so the /events page *should* show only de-duped event information. 

And discussed on Slack, we do need the splat pages for reporting event information to partners.

File: bin/cron/hoc_signup_counts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+PEGASUS_REPORTING_DB_READONLY = sequel_connect(
+  CDO.pegasus_reporting_db_reader,
+  CDO.pegasus_reporting_db_reader
+)
+
+def main
+  year = (ARGV[0] || DCDO.get('hoc_year', DEFAULT_YEAR)).to_i
+
+  # Note that country_counts and us_state_counts are of the form
+  # [{"country" => nil, "count" => 1}, {"country" => "Afghanistan", "count" => 2}, ...].
```
Comment (breville) [inline bin/cron/hoc_signup_counts:22]: Curious.. what does country of `nil` mean?  And what does a US state one look like?

**Comment by breville** [comment]
Does this also deal with the cities counts that can be seen in cdo/properties.rb?

File: bin/cron/hoc_signup_counts (hunk 1 above)
Comment (ashercodeorg) [reply bin/cron/hoc_signup_counts:22]: A country of `nil` means the geocoder was not able to determine a country for the IP address. The `"count"` values are the number of HOC events (signups) in that region, so `{"state" => "Washington", "count" => 1}` represents one HOC event in the state of Washington.

**Comment by ashercodeorg** [comment]
Let me change the logic in this PR to dedupe by email address, confirming that those numbers match the SOLR numbers.

**Comment by ashercodeorg** [comment]
Merging, as offline discussion suggests the data deduping should happen at the data level rather than the analysis level. Regardless of whether we can completely sever the SOLR dependency, it seems worthwhile to make use of this type of thing wherever we can.

**Pull request merged**